code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm

```diff
@@ -302,7 +302,11 @@ GLOBAL_LIST_EMPTY_TYPED(air_alarms, /obj/machinery/airalarm)
 			var/obj/machinery/atmospherics/components/powering = vent || scrubber
 			powering.on = !!params["val"]
 			powering.update_appearance(UPDATE_ICON)
-			powering.check_atmos_process()
+			//special check for scrubbers since its turf determines if it should process
+			if(isnull(scrubber))
+				vent.check_atmos_process()
```

Comment:
this would never be valid, it needs the args to know what to check off

Reply:
So exact same as scrubber then.  Do we move the try_update_atmos_process() to the parent type so both vent and scrubber share the same proc?

Reply:
Ok i'm even more confused.

vents don't override the `should_atmos_process()` proc so `check_atmos_process()` will never run so why even bother?

Reply:
Poor proc names are confusing you here (just now saw this sorry for not responding for so long)

`should_atmos_process()` is a part of `/datum/element/atmos_sensitive`.
The element as a whole allows objects in the world to efficiently react to air changing in the environment around them.

if `should_atmos_process()` returns TRUE, then `atmos_expose()` is called on the object once every SSair fire until the `should` proc returns false.

Scrubbers use them because they are fully event based, they'll only start scrubbing if the air in front of them is at a particular pressure.
Vents don't, because they aren't event based yet. This means a lot of passive cpu cost checking to see if vents around the station can push air.

You are confusing this atmos_sensitive system with `process_atmos()`, which is similar, but not based around reacting to the air around the machines.

The correct solution here is close to what you're doing, the helper proc is good, but you don't need to override `should_atmos_process()` at all. Just let vents ignore the call